framework/include/cppmicroservices/BundleContext.h

```diff
@@ -123,6 +123,11 @@ namespace cppmicroservices
          */
         BundleContext();
 
+        /**
+         * Allocates a new %BundleContext object that is a copy of an existing one.
+         */
+        BundleContext* NewBundleContext();
```

Comment:
I'm concerned about inappropriate use of this method by users. This method has a very specific purpose and it should never be used by cppmicroservices users. Can we guard this somehow or put up some indication to users that they should never call this themselves?

Reply:
A clear comment of intent will have to do for now.